handlers/soa_resource.go

```diff
@@ -111,3 +111,36 @@ func HandleCreateStatementOfAffairs(svc dao.Service) http.Handler {
 		utils.WriteJSONWithStatus(w, req, transformers.StatementOfAffairsDaoToResponse(statementDao), http.StatusOK)
 	})
 }
+
+// HandleGetStatementOfAffairs retrieves a statement of affairs stored against the Insolvency Case
+func HandleGetStatementOfAffairs(svc dao.Service) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		vars := mux.Vars(req)
+		transactionID := utils.GetTransactionIDFromVars(vars)
+		if transactionID == "" {
+			log.ErrorR(req, fmt.Errorf("there is no transaction ID in the URL path"))
+			m := models.NewMessageResponse("transaction ID is not in the URL path")
+			utils.WriteJSONWithStatus(w, req, m, http.StatusBadRequest)
+			return
+		}
+
+		log.InfoR(req, fmt.Sprintf("start GET request for get statement of affairs with transaction id: %s", transactionID))
+
+		statementOfAffairs, err := svc.GetStatementOfAffairsResource(transactionID)
+		if err != nil {
+			log.ErrorR(req, fmt.Errorf("failed to get statement of affairs from insolvency resource in db for transaction [%s]: %v", transactionID, err))
+			m := models.NewMessageResponse("there was a problem handling your request")
+			utils.WriteJSONWithStatus(w, req, m, http.StatusInternalServerError)
+			return
+		}
+		if statementOfAffairs.StatementDate == "" {
+			m := models.NewMessageResponse("statement of affairs not found on transaction")
```

Comment:
Should we pass in the transaction ID as well?

```suggestion
			m := models.NewMessageResponse(fmt.Sprintf("statement of affairs not found on transaction with ID: [%s]", transactionID))
```